src/herbivoresAndCarnivores.js

```diff
@@ -1,35 +1,43 @@
 'use strict';
-
 class Animal {
   static alive = [];
+  static DEFAULT_HEALTH = 100;
 
-  constructor(name, health = 100) {
+  constructor(name, health = Animal.DEFAULT_HEALTH) {
     this.name = name;
     this.health = health;
-    this.constructor.alive.push(this);
+    Animal.alive.push(this);
   }
 
-  static removeAnimal() {
+  static removeDeadAnimals() {
     this.alive = this.alive.filter((animal) => animal.health > 0);
   }
 }
 
 class Herbivore extends Animal {
-  constructor(name, health = 100) {
+  static DAMAGE_FROM_BITE = 50;
+
+  constructor(name, health = Animal.DEFAULT_HEALTH) {
     super(name, health);
     this.hidden = false;
   }
 
   hide() {
     this.hidden = true;
   }
+
+  receiveBite() {
+    if (!this.hidden) {
+      this.health -= Herbivore.DAMAGE_FROM_BITE;
+      Animal.removeDeadAnimals();
```

Comment:
We can check if `this.health <= 0` and call `Animal.removeDeadAnimals()` only if that's true - to avoid redundant calls of that method